### PR TITLE
Updating machines through webscraping

### PR DESCRIPTION
**Pull request by jannisborn**
Partly addresses #36

### CODE REVIEW

File: backend/scripts/location_differ.py [hunk 1]
```diff
@@ -162,15 +173,22 @@ def location_differ(
                             # Easy case, we just add this machine to server_dict
                             if len(device_dict[this_link]) > 1:
```
Comment (NinaWie) [inline backend/scripts/location_differ.py:174]: Move check down to server case bc it cant happen here

File: backend/scripts/location_differ.py [hunk 2]
```diff
@@ -230,10 +254,90 @@ def location_differ(
...
 
+            # Verify that machine is indeed new through fuzzy search
+            match, score = fuzzysearch.extract(
+                this_title, list(no_link["name"]), limit=1
+            )[0]
+            m_idx = list(no_link["name"]).index(match)
+
+            if (
+                score > 87
+                and geojson["properties"]["area"] == no_link.loc[m_idx]["area"]
```
Comment (NinaWie) [inline backend/scripts/location_differ.py:270]: better to filter for area before the search

File: backend/scripts/location_differ.py [hunk 3]
```diff
@@ -230,10 +254,90 @@ def location_differ(
...
+                        f"Seeems that machine {this_title} already exists as: {match}"
+                    )
+                    # Identify which machine this is in server_data and then update it
+                    assert (
+                        server_data["features"][m_idx]["properties"]["external_url"]
+                        == "null"
+                    )
+                    server_data["features"][idx]["properties"][
+                        "external_url"
+                    ] = this_link
```
Comment (NinaWie) [inline backend/scripts/location_differ.py:298]: idx not set

File: backend/scripts/location_differ.py [hunk 4]
```diff
@@ -230,10 +254,90 @@ def location_differ(
...
+                        "logs"
+                    ] = f"Distance {dist} km to {server_data['features'][idx]['properties']['id']}"
+                    problem_data["features"].append(geojson)
+                else:
+                    logger.info(
+                        f"Seeems that machine {this_title} already exists as: {match}"
+                    )
+                    # Identify which machine this is in server_data and then update it
+                    assert (
+                        server_data["features"][m_idx]["properties"]["external_url"]
```
Comment (NinaWie) [inline backend/scripts/location_differ.py:293]: wrong list index

File: backend/scripts/location_differ.py [hunk 5]
```diff
@@ -244,7 +348,7 @@ def location_differ(
             lat, lng = get_coordinates(
                 title=geojson["properties"]["name"],
                 subtitle=geojson["properties"]["address"],
-                api_key=api_key,
+                api=gmaps,
```
Comment (NinaWie) [inline backend/scripts/location_differ.py:351]: do we need the temporary?

**Review by NinaWie — CHANGES_REQUESTED** [review]

File: backend/scripts/location_differ.py (hunk 5 above)
Comment (jannisborn) [reply backend/scripts/location_differ.py:351]: We do since the `geojson` is a prelim object

File: backend/scripts/location_differ.py (hunk 1 above)
Comment (jannisborn) [reply backend/scripts/location_differ.py]: Done

File: backend/scripts/location_differ.py (hunk 3 above)
Comment (jannisborn) [reply backend/scripts/location_differ.py]: Done

File: backend/scripts/location_differ.py (hunk 4 above)
Comment (jannisborn) [reply backend/scripts/location_differ.py]: Fixed

**Pull request merged**